homeassistant/components/proximity/__init__.py

```diff
@@ -121,33 +121,34 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     await coordinator.async_config_entry_first_refresh()
     hass.data[DOMAIN][entry.entry_id] = coordinator
 
-    # legacy proximity entity handling, can be removed in 2024.8
-    proximity = Proximity(hass, entry.title, coordinator)
-    await proximity.async_added_to_hass()
-    proximity.async_write_ha_state()
+    async def _asnyc_setup_legacy() -> None:
```

Comment:
```suggestion
    async def _async_setup_legacy() -> None:
```

Reply:
I'd move it to the module level and accept hass, config_entry and coordinator.

Reply:
11e3df0a33323a2d20422bfbb7a8bb4811f41ef2